presto-main/src/main/java/com/facebook/presto/execution/buffer/LazyOutputBuffer.java

```diff
@@ -206,6 +206,19 @@ public void setOutputBuffers(OutputBuffers newOutputBuffers)
         return outputBuffer.get(bufferId, token, maxSize);
     }
 
+    @Override
+    public void acknowledge(OutputBufferId bufferId, long token)
+    {
+        OutputBuffer outputBuffer;
+        synchronized (this) {
+            if (delegate == null) {
+                return;
```

Comment:
maybe throw here... this should never happen and is the sign of a programming error

Reply:
Use `checkState` since that's the right exception